ansible_runner/runner_config.py

```diff
@@ -270,6 +281,60 @@ def generate_ansible_command(self):
 
         return exec_list
 
+    def build_process_isolation_temp_dir(self):
+        '''
+        Create a temporary directory for process isolation to use.
+        '''
+        path = tempfile.mkdtemp(prefix='ansible_runner_pi_', dir=self.process_isolation_path)
+        os.chmod(path, stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR)
+        return path
+
+    def wrap_args_with_process_isolation(self, args):
+        '''
+        Wrap existing command line with proot to restrict access to:
+         - self.process_isolation_path (generally, /tmp) (except for own /tmp files)
+        '''
+        cwd = os.path.realpath(self.cwd)
+        pi_temp_dir = self.build_process_isolation_temp_dir()
+        new_args = [self.process_isolation_executable or 'bwrap', '--unshare-pid', '--dev-bind', '/', '/', '--proc', '/proc']
+
+        for path in sorted(set(self.process_isolation_hide_paths or [])):
+            if not os.path.exists(path):
+                continue
+            path = os.path.realpath(path)
+            if os.path.isdir(path):
+                new_path = tempfile.mkdtemp(dir=pi_temp_dir)
+                os.chmod(new_path, stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR)
+            else:
+                handle, new_path = tempfile.mkstemp(dir=pi_temp_dir)
+                os.close(handle)
+                os.chmod(new_path, stat.S_IRUSR | stat.S_IWUSR)
+            new_args.extend(['--bind', '%s' %(new_path,), '%s' % (path,)])
+
+        if self.private_data_dir:
+            show_paths = [self.private_data_dir]
+        else:
+            show_paths = [cwd]
+
+        if self.process_isolation_venv:
+            new_args.extend(['--ro-bind', self.process_isolation_venv, self.process_isolation_venv])
+
+        show_paths.extend(self.process_isolation_show_paths or [])
+        for path in sorted(set(show_paths)):
+            if not os.path.exists(path):
+                continue
+            path = os.path.realpath(path)
```

Comment:
:+1: I tested `os.path.exists` behavior and `os.path.realpath` behavior. Both resolve symlinks.

```
lrwxrwxrwx   1 meyers meyers        4 Nov 16 08:12 dir1 -> dir2
drwxr-xr-x   2 meyers meyers     4096 Nov 16 08:12 dir2
lrwxrwxrwx   1 meyers meyers        4 Nov 16 08:12 dir3 -> dir4
```

```
>>> import os
>>> os.path.exists('/tmp/dir1')
True
>>> os.path.exists('/tmp/dir3')
False
>>> os.path.realpath('/tmp/dir3/')
'/tmp/dir4'
```